src/integrationTest/java/uk/gov/hmcts/darts/audio/controller/AudioControllerAddAudioNoSessionIntTest.java

```diff
@@ -0,0 +1,174 @@
+import uk.gov.hmcts.darts.testutils.IntegrationBase;
+import uk.gov.hmcts.darts.testutils.stubs.AuthorisationStub;
+import uk.gov.hmcts.darts.testutils.stubs.SuperAdminUserStub;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.multipart;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@AutoConfigureMockMvc
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@TestPropertySource(properties = {
+    "spring.servlet.multipart.max-file-size=4MB",
+    "spring.servlet.multipart.max-request-size=4MB",
+})
+@SuppressWarnings({"PMD.DoNotUseThreads", "PMD.AvoidInstantiatingObjectsInLoops", "PMD.AvoidThrowingRawExceptionTypes"})
+class AudioControllerAddAudioNoSessionIntTest extends IntegrationBase {
+
+    @Value("${local.server.port}")
+    protected int port;
+
+    private static final URI ENDPOINT = URI.create("/audios");
+    private static final OffsetDateTime STARTED_AT = OffsetDateTime.of(2024, 10, 10, 10, 0, 0, 0, ZoneOffset.UTC);
+    private static final Path AUDIO_BINARY_PAYLOAD_1 = DataGenerator.createUniqueFile(DataSize.ofBytes(10),
+                                                                                      DataGenerator.FileType.MP2);
+
+
+    @Autowired
+    private MockMvc mockMvc;
+    @Autowired
+    private AuthorisationStub authorisationStub;
+    @MockBean
+    private UserIdentity mockUserIdentity;
+
+
+    @Autowired
+    private SuperAdminUserStub superAdminUserStub;
+
+
+    @BeforeEach
+    void beforeEach() {
+        UserAccountEntity testUser = authorisationStub.getTestUser();
+        when(mockUserIdentity.getUserAccount()).thenReturn(testUser);
+
+        dartsDatabase.createCourthouseUnlessExists("Bristol");
+
+    }
+
+
+    @Test
+    void testAddAudioWithConcurrency() throws Exception {
+        superAdminUserStub.givenUserIsAuthorised(mockUserIdentity, SecurityRoleEnum.MID_TIER);
+
+        int numberOfThreads = 5;
+        try (ExecutorService service = Executors.newFixedThreadPool(5)) {
+            CountDownLatch latch = new CountDownLatch(numberOfThreads);
+            List<Future<?>> futures = new ArrayList<>();
+
+            for (int i = 0; i < numberOfThreads; i++) {
+                final int threadNum = i;
+                Future<?> future = service.submit(() -> {
```

Comment:
Runnable could be defined outside of the loop and passed in as a var instead of instantiating a new runnable each time

Reply:
my runnable is different every time, they are using different dates for the audio requests.

Reply:
Aha I missed that :)